.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+# Workflow label
+name: CI
+
+# Workflow trigger
+on: [push, pull_request]
+
+# Workflow tasks
+jobs:
+  # Apply lint, check formatting
+  lint:
+    name: "Lint (Python ${{ matrix.python-version }})"
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Cache Python
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{matrix.os}}-${{matrix.python-version}}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}
+      - name: Install Python requirements
+        run: pip install --upgrade --upgrade-strategy eager .[dev]
+
+      - name: Lint with flake8
+        run: |
+          pip install flake8
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+        shell: bash
+
+      - name: Cancelling pipeline (failed)
+        if: failure()
+        uses: andymckay/cancel-action@0.2
+
+      - name: Check formatting with black
+        run: black --check .
```

Comment:
Move these lines above the cancellation step

Reply:
I moved them down since we don't want the CI to stop if the code is not formatted accordingly to black, we just want a notification

Reply:
I find it a bit weird..

Reply:
I would like CI to finish with the remaining checks. Stopping just because the code is not beautiful should not be our goal...

Reply:
Note that a possibility is also to automatically fix formatting by pushing formatted files back: https://github.com/psf/black/actions/runs/17913292/workflow